data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/KafkaCustomProducerFactory.java

```diff
@@ -56,21 +57,23 @@ public KafkaCustomProducerFactory(
     public KafkaCustomProducer createProducer(final KafkaProducerConfig kafkaProducerConfig,
                                               final ExpressionEvaluator expressionEvaluator, final SinkContext sinkContext, final PluginMetrics pluginMetrics,
                                               final DLQSink dlqSink,
+                                              AtomicInteger maxRequestSize,
```

Comment:
Mutating an existing object is not a very maintainable practice.

Can you add the `maxRequestSize` as a getter on the `KafkaCustomProducer` which this will return?